tests/unit/project_loader/extensions/test_gnome_3_34.py

```diff
@@ -0,0 +1,418 @@
+                        },
+                        {
+                            "PKG_CONFIG_PATH": "/snap/gnome-3-34-1804-sdk/current/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/pkgconfig:/snap/gnome-3-34-1804-sdk/current/usr/lib/pkgconfig:/snap/gnome-3-34-1804-sdk/current/usr/share/pkgconfig:$PKG_CONFIG_PATH"
+                        },
+                        {
+                            "GETTEXTDATADIRS": "/snap/gnome-3-34-1804-sdk/current/usr/share/gettext-current:$GETTEXTDATADIRS"
+                        },
+                        {
+                            "GDK_PIXBUF_MODULE_FILE": "/snap/gnome-3-34-1804-sdk/current/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/gdk-pixbuf-current/loaders.cache"
+                        },
+                    ]
+                }
+            ),
+        )
+        self.expectThat(
+            gnome_extension.parts,
+            Equals(
+                {
+                    "gnome-3-34-extension": {
+                        "source": "$SNAPCRAFT_EXTENSIONS_DIR/desktop",
+                        "source-subdir": "gnome",
+                        "plugin": "make",
+                        "build-snaps": ["gnome-3-34-1804-sdk/latest/stable"],
+                    }
+                }
+            ),
+        )
+
+    def test_supported_bases(self):
+        self.assertThat(ExtensionImpl.get_supported_bases(), Equals(("core18",)))
+
+    def test_supported_confinement(self):
+        self.assertThat(
+            ExtensionImpl.get_supported_confinement(), Equals(("strict", "devmode"))
+        )
+
+    def test_no_defined_build_env(self):
+        """
+        Test that the build-environment defined by the gnome-3-34 extension is
+        applied to all parts. In this test case, there is no user-defined
+        build-environment section in any part.
+        """
+        self.make_snapcraft_yaml(
+            textwrap.dedent(
+                """\
+                name: test
+                version: '1'
+                summary: test
+                description: test
+                base: core18
+                grade: stable
+                confinement: strict
+
+                apps:
+                    test-app:
+                        command: echo "hello"
+                        extensions: [gnome-3-34]
+
+                parts:
+                    test-part-1:
+                        plugin: nil
+                    test-part-2:
+                        plugin: nil
+                """
+            )
+        )
+
+        result = self.run_command(["expand-extensions"])
+
+        self.assertThat(
+            result.output,
+            Contains(
+                textwrap.dedent(
+                    """\
+                        parts:
+                          test-part-1:
+                            plugin: nil
+                            build-environment: &id001
```

Comment:
Hrmm, I wonder what this `&id001` is about...?

Reply:
it's referencing the first build-environment section. Since we didn't give it a name, one was assigned to it: "id001".

A similar thing is done in [the build-snap](https://gitlab.gnome.org/Community/Ubuntu/gnome-sdk/blob/gnome-3-34-1804-sdk/snapcraft.yaml) where the builld-environment section is used in every part, but only explicitly defined in the first part.